tests/scripts/all.sh

```diff
@@ -2491,6 +2491,7 @@ psa_crypto_config_accel_all_curves_except_one () {
     scripts/config.py unset MBEDTLS_ECP_RESTARTABLE
 
     # disable modules for which we have drivers
+    scripts/config.py unset MBEDTLS_ECP_C
```

Comment:
I'm confused. Don't we already have `component_test_psa_crypto_config_accel_all_ec_algs_use_psa` for that?

This PR is not just refactoring existing code in existing configs, so it shouldn't need new tests. (It's preparation work for making new configs possible in the future, namely enabling PK in `config_psa_crypto_full_all_ec_algs_no_ecp_use_psa`, but that will come later one 7074 is also done, and perhaps a small follow-up after that.)

Reply:
This problem was raised by a CI failure with `test_psa_crypto_config_accel_all_curves_except_[p192/x25519]`. The error was in `pk_wrap.c`: it was picking the "wrong" the `ecdsa_verify_wrap`, the one using `mbedtls_ecdsa_read_signature`. This happened because, following our discussion(1), I changed the guards for that function from `USE_PSA_CRYPTO` to `MBEDTLS_PK_USE_PSA_EC_PUB_KEY`. The latter is only available when `!ECP_C`.

A part from having the CI happy, to me this change seems reasonable according to the fact that `ECP_LIGHT` was only introduced after these failing test cases. However, as far as they are concerned, we don't want the full `ECP_C` here, but only `ECP_LIGHT`. Wdyt?

(1) i.e. when `ECP_C` is defined we have to use the legacy ecp_keypair structure and legacy functions..

Reply:
> (1) i.e. when `ECP_C` is defined we have to use the legacy ecp_keypair structure and legacy functions..

I think there's a misunderstanding: when `USE_PSA` crypto is defined, we have to use PSA for crypto operations (except when that is not possible, which currently is only restartable operations). So, the decision tree is:
- if `!USE_PSA`, use legacy context and legacy functions (unchanged)
- if `USE_PSA`, use PSA functions, but:
  - if `ECP_C`: keep the key material in `ecp_keypair` and convert to PSA before each operation (also unchanged, except perhaps [re-organizing the code in order to share code with the next case](https://github.com/Mbed-TLS/mbedtls/issues/7570#issuecomment-1545520593))
  - if `!ECP_C`: we can keep the key material in new PSA-friendly fields to avoid conversions (that's new).

Again, this is a refactoring PR, configurations that used to work are supposed to keep working the same way.

Note that in [the code structure I suggested](https://github.com/Mbed-TLS/mbedtls/issues/7570#issuecomment-1545498889), in the `USE_PSA` case we are using PSA all the time and the only thing that varies depending on config (`ECP_C` or not) and key type (opaque or not, public or private) is how we transform (or not) the key material from the form we have to the form used by the internal function - but that function always uses PSA.

Reply:
Ok, so I think I'll have to restructure the commits because I'm not sure I correctly addressed this point:

> if USE_PSA, use PSA functions, but:
    if ECP_C: keep the key material in ecp_keypair and convert to PSA before each operation (also unchanged, except perhaps https://github.com/Mbed-TLS/mbedtls/issues/7570#issuecomment-1545520593)

I'll try to do it asap

Reply:
@mpg I think I reshaped the code as you suggested and indeed the issue causing this problem/comment is not present any more. Since the CI is also happy, I think this PR is ready for a review ;)